src/zc/buildout/tests.py

```diff
@@ -3062,6 +3062,39 @@ def test_abi_tag_eggs():
     ...
     """
 
+def test_buildout_doesnt_keep_adding_itself_to_versions():
+    r"""
+    We were constantly writing to versions.cfg for buildout and setuptools
+
+    >>> write('buildout.cfg',
+    ... '''
+    ... [buildout]
+    ... parts =
+    ... extends = versions.cfg
+    ... show-picked-versions = true
+    ... update-versions-file = versions.cfg
+    ... extends = versions.cfg
+    ... ''')
+    >>> write('versions.cfg',
+    ... '''[versions]
+    ... ''')
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions = f.read()
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> _ = system(join('bin', 'buildout'))
+    >>> with open('versions.cfg') as f:
+    ...     versions == f.read()
+    True
+    >>> cat('versions.cfg') # doctest: +ELLIPSIS
```

Comment:
I assume there's no risk of ELLIPSIS hiding extra lines getting appended to the file.